static/app/views/issueList/customViewsHeader.tsx

```diff
@@ -300,51 +300,56 @@ function CustomViewsIssueListHeaderTabsContent({
 
   // Update local tabs when new views are received from mutation request
   useEffectAfterFirstRender(() => {
-    // Find views with ids that are not in the current tabs
+    const assignedIds = new Set();
```

Comment:
One more thing - can you instantiate this Set inside the `setDraggableTags` setter? State setters should be idempotent because react might call them multiple times so we don't want to mutate anything outside the setter.

Might be better to use the callback version anyway so you know you're getting the right `draggableTabs` state:

```tsx
setDraggableTags(oldDraggableTags => {
  // ...
})
```

Reply:
Oh good call I didn't know that, changing that now!